src/action.js

```diff
@@ -169,7 +203,7 @@ async function main() {
     );
   }
 
-  if (isPullRequestEvent(githubEventName)) {
+  if (isPullRequestEvent(githubEventName) && !needsInsights(config)) {
```

Comment:
issue(blocking): `needsInsights` is `async` so we need to make sure we `await` it